scripts/process_data.py

```diff
@@ -110,40 +143,111 @@ def convert_video_to_images(
+    CONSOLE.log("[bold green]:tada: Done converting video to images.")
+
+    return summary_log
+
+
+def convert_insta360_to_images(
+    video_front: Path,
+    video_back,
+    image_dir: Path,
+    num_frames_target: int,
+    crop_percentage: float = 0.7,
+    verbose: bool = False,
+) -> List[str]:
+    """Converts a video into a sequence of images.
 
-    ffmpeg_cmd += f" {out_filename}"
+    Args:
+        video_front: Path to the front video.
+        video_back: Path to the back video.
+        output_dir: Path to the output directory.
+        num_frames_target: Number of frames to extract.
+        verbose: If True, logs the output of the command.
+    Returns:
+        A summary of the conversion.
+    """
 
-    run_command(ffmpeg_cmd, verbose=verbose)
+    with status(msg="Converting video to images...", spinner="bouncingBall", verbose=verbose):
+        # delete existing images in folder
+        for img in image_dir.glob("*.png"):
+            if verbose:
+                CONSOLE.log(f"Deleting {img}")
+            img.unlink()
+
+        num_frames_front = get_num_frames_in_video(video_front)
+        num_frames_back = get_num_frames_in_video(video_back)
+        if num_frames_front == 0:
+            CONSOLE.print(f"[bold red]Error: Video has no frames: {video_front}")
+            sys.exit(1)
+        if num_frames_back == 0:
+            CONSOLE.print(f"[bold red]Error: Video has no frames: {video_front}")
+            sys.exit(1)
+
+        spacing = num_frames_front // (num_frames_target // 2)
+        vf_cmds = []
+        if spacing > 1:
+            vf_cmds = [f"thumbnail={spacing}", "setpts=N/TB"]
+        else:
+            CONSOLE.print("[bold red]Can't satify requested number of frames. Extracting all frames.")
 
-    return num_frames, len(list(image_dir.glob("*.png")))
+        vf_cmds.append(f"crop=iw*({crop_percentage}):ih*({crop_percentage})")
+
+        front_vf_cmds = vf_cmds + ["transpose=2"]
+        back_vf_cmds = vf_cmds + ["transpose=1"]
+
+        front_ffmpeg_cmd = (
+            f"ffmpeg -i {video_front} -vf '{','.join(front_vf_cmds)}' -r 1 {image_dir / 'frame_%05d.png'}"
+        )
+        back_ffmpeg_cmd = (
+            f"ffmpeg -i {video_back} -vf '{','.join(back_vf_cmds)}' -r 1 {image_dir / 'back_frame_%05d.png'}"
+        )
+
+        run_command(front_ffmpeg_cmd, verbose=verbose)
+        run_command(back_ffmpeg_cmd, verbose=verbose)
+
+        num_extracted_front_frames = len(list(image_dir.glob("frame*.png")))
+        for i, img in enumerate(image_dir.glob("back_frame_*.png")):
+            img.rename(image_dir / f"frame_{i+1+num_extracted_front_frames:05d}.png")
+
+    summary_log = []
+    summary_log.append(f"Starting with {num_frames_front + num_frames_back} video frames")
+    summary_log.append(f"We extracted {len(list(image_dir.glob('*.png')))} images")
+    CONSOLE.log("[bold green]:tada: Done converting insta360 to images.")
```

Comment:
Why not just a long string? Instead of a List.

Reply:
The log is something that continually gets appended to throughout the script. I find it a bit cleaner to keep as a list.